### PR TITLE
Adds M07 Training Grenade

### PR DESCRIPTION
**Pull request by TeaTotalTemplar**
## About the PR
- Adds the M07 Training grenade, a safe version of the M40 HEDP with no explosive charge.
- Grenade functions identically to M40 HEDP and can be placed into grenade launchers or thrown by hand.
- Placeholder sound effect due to CM13 version using motiondetector sound effects which may? be under copyright

## Why / Balance

CM13 Parity

## Media
image

## Requirements

- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**
🆑 
- add: Adds the M07 Training Grenade - this grenade functions identically to the M40 HEDP and HEFA but contains no explosive charge, allowing new recruits to familiarise themselves with the M40 in a safe environment. Can be loaded into grenade launchers, or thrown by hand.

### CODE REVIEW

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Throwable/grenades.yml [hunk 1]
```diff
@@ -158,6 +158,23 @@
...
+  components:
+  - type: Sprite
+    sprite: _RMC14/Objects/Weapons/Grenades/m07training.rsi
+  - type: SoundOnTrigger
+    sound: /Audio/Effects/snap.ogg
+  - type: Tag
+    tags:
+    - Grenade
+    - LauncherCompatibleGrenade
+    - RMCGrenadeTraining
```
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Throwable/grenades.yml:175]: Add the m40 grenade tag on here, it should make it conpatiable with all the launchers in game

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Attachments/under_attachments.yml [hunk 2]
```diff
@@ -347,6 +347,7 @@
     whitelist:
       tags:
       - RMCGrenadeM40
+      - RMCGrenadeTraining
```
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Attachments/under_attachments.yml:350]: You can drop this tag whitelist if you add grenade m40 tag to the training grenade

**Review by QuietlyWhisper — CHANGES_REQUESTED** [review]
Minor fix

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Attachments/under_attachments.yml (hunk 2 above)
Comment (TeaTotalTemplar) [reply Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Attachments/under_attachments.yml]: done, thank you

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**